Apps/WebClient/src/Server/Services/UserEmailService.cs

```diff
@@ -130,7 +144,17 @@ public bool UpdateUserEmail(string hdid, string email, Uri hostUri, string beare
             if (!string.IsNullOrEmpty(email))
             {
                 this.logger.LogInformation($"Sending new email invite for user ${hdid}");
-                this.emailQueueService.QueueNewInviteEmail(hdid, email, hostUri);
+
+                Guid? existingInviteKey = null;
+                if (emailInvite != null
+                    && emailInvite.Email != null
+                    && !string.IsNullOrEmpty(emailInvite.Email.To)
+                    && email.Equals(emailInvite.Email.To, StringComparison.OrdinalIgnoreCase))
+                {
+                    existingInviteKey = emailInvite.InviteKey;
+                }
```

Comment:
this section has the smarts to generate a new key or not. 
```
else { 
   existingInviteKey = new Guid();
}

Reply:
updated